### DISCUSSION TITLE
iOS PAG 播放资源异常日志

### DISCUSSION THREAD

**Discussion by piglikeYoung** [Tencent/libpag#2285] [Q&A] [answered]
平台：iOS 原生
问题：目前 PAG 加载资源或播放出错后，SDK 的接口没有任何异常接口给到调用方，请问下有什么办法能知晓这些异常吗？
现有 PAGViewListener 监听，但是也没有 error 参数返回

**Comment by kevingpqi123** [comment, marked as answer]
可以通过 PAGFile 的判断是否为空来确定 pag 资源是否收到损坏
如果是播放层面的问题，如果加载失败或播放出错就是必现的 bug了，这些是需要 pag 修复的，没有异常可以抛出，有类似问题直接反馈给我们就好，对于业务侧而言，有 PAGViewListener 就够了，整个播放过程中都可以监听到

**Reply by piglikeYoung** [reply]
PAG 有两种加载方式
- pagView.setComposition(pagFile)
- pagView.setPath(path)

PAGFile 只有 setComposition 可以通过 PAGFile 识别，setPath 获取不到 PAGFile

**Reply by kevingpqi123** [reply]
The duration of current composition in microseconds.

- (int64_t)duration
Declared In
PAGView.h

可以使用这个接口，建议系统了解下相关文档

**Reply by piglikeYoung** [reply]
意思是使用 pagView.setPath(path) 接口的，可以通过获取 duration 来确认资源加载是否成功？

**Reply by kevingpqi123** [reply]
是的，如果值 > 0, 就会有数据，通过 setComposition 不想提前判断，也可以使用这个接口
